tests/test_commenter.py

```diff
@@ -0,0 +1,336 @@
+    return {
+        "job_id": 1003,
+        "name": "test_running",
+        "status": "running",
+    }
+
+
+@pytest.fixture
+def mock_job_failed() -> dict:
+    return {
+        "job_id": 1003,
+        "name": "test_failed",
+        "status": "failed",
+    }
+
+
+@pytest.fixture
+def mock_job_passed() -> dict:
+    return {
+        "job_id": 1003,
+        "name": "test_passed",
+        "status": "passed",
+    }
+
+
+@pytest.fixture
+def mock_incident_smelt() -> Mock:
+    mock_incident = Mock(spec=Incident)
+    mock_incident.id = 1
+    mock_incident.type = "smelt"
+    mock_incident.revisions = {}
+    return mock_incident
+
+
+@pytest.fixture
+def mock_incident_smelt_with_revisions() -> Mock:
+    mock_incident = Mock(spec=Incident)
+    mock_incident.id = 2
+    mock_incident.type = "smelt"
+    mock_incident.revisions = {
+        ArchVer("x86_64", "15-SP4"): 12345,
+        ArchVer("aarch64", "15-SP4"): 12346,
+    }
+    return mock_incident
+
+
+@pytest.fixture
+def mock_commentapi() -> Mock:
+    mock_commentapi = Mock()
+    mock_commentapi.get_comments.return_value = []
+    mock_commentapi.comment_find.return_value = (None, None)
+    mock_commentapi.add_marker.return_value = "Marked message"
+    mock_commentapi.truncate.return_value = "Marked message"
+    return mock_commentapi
+
+
+def test_commenter_init(mock_args: Namespace, mocker: MockerFixture) -> None:
+    mock_client = mocker.patch("openqabot.commenter.openQAInterface")
+    mocker.patch("openqabot.commenter.get_incidents", return_value=[MagicMock()])
+    mocker.patch("openqabot.commenter.CommentAPI")
+    mocker.patch("openqabot.commenter.osc.conf.get_config")
+    c = Commenter(mock_args)
+
+    assert c.dry
+    assert c.token == {"Authorization": "Token test_token"}
+    assert c.client == mock_client.return_value
+
+
+def test_commenter_call(mock_args: Namespace, caplog: pytest.LogCaptureFixture, mocker: MockerFixture) -> None:
+    caplog.set_level(logging.DEBUG)
+    mock_incident = Mock(spec=Incident)
+    mock_incident.id = 1
+    mock_incident.type = "maintenance"
+
+    mocker.patch("openqabot.commenter.openQAInterface")
+    mocker.patch("openqabot.commenter.get_incidents", return_value=[mock_incident])
+    mocker.patch("openqabot.commenter.osc.conf.get_config")
+    mocker.patch("openqabot.commenter.CommentAPI")
+
+    c = Commenter(mock_args)
+    ret = c()
+
+    assert ret == 0
+    assert "Skipping incident 1 of type maintenance" in caplog.text
+
+
+def test_commenter_call_value_error_incident_results(
+    mocker: MockerFixture, mock_args: Namespace, mock_incident_smelt: Mock, caplog: pytest.LogCaptureFixture
+) -> None:
+    caplog.set_level(logging.DEBUG)
+
+    mocker.patch("openqabot.commenter.openQAInterface")
+    mocker.patch("openqabot.commenter.get_incidents", return_value=[mock_incident_smelt])
+    mocker.patch("openqabot.commenter.osc.conf.get_config")
+    mocker.patch("openqabot.commenter.CommentAPI")
```

Comment:
Those seem to be reused multiple times. I suggest to extract that into a fixture possibly taking mock_incident_smelt.rr as argument?